### PR TITLE
JSON load error handling.

### PR DESCRIPTION
**Pull request by ashcrow**
Adds `try`/`except` around json loading.

### CODE REVIEW

File: homu/main.py [hunk 1]
```diff
@@ -863,8 +869,14 @@ def main():
...
-        with open('cfg.json') as fp:
-            cfg = json.loads(fp.read())
+        try:
+            with open('cfg.json') as fp:
+                try:
+                    cfg = json.loads(fp.read())
+                except ValueError as ve:
+                    logger.fatal('Malformed json in config.')
+        except FileNotFoundError:
+            raise RuntimeError('No configuration provided')
```
Comment (cgwalters) [inline homu/main.py]: "...provided in cfg.toml or cfg.json" maybe?

File: homu/server.py [hunk 2]
```diff
@@ -439,8 +445,15 @@ def buildbot():
...
+    packets = None
+    try:
+        packets = json.loads(request.forms.packets)
+    except ValueError as ve:
+        logger.error('/buildbot packets was not valid json.')
+        lazy_debug(logger, lambda: 'invalid json: {}'.format(request.forms.packets))
+        abort(400, 'Bad Packets')
 
-    for row in json.loads(request.forms.packets):
+    for row in packets:
```
Comment (cgwalters) [inline homu/server.py:456]: Won't this traceback in the error case if `packets = None`?

Comment (cgwalters) [inline homu/server.py:456]: Nevermind, I read what `abort` does.

**Comment by cgwalters** [comment]
Looks reasonable to me.  One minor style comment; I saw in another project you use `.` in commit message subjects, but that's (mostly) not the style of this project, see e.g.: https://github.com/barosl/homu/commit/0ce1a365f751c6350abfe547b40cef2e0493c059

Also...did you hit this in practice?  It'd be interesting to know that if true, something like "Travis sent corrupted JSON once" in the commit message.  Or if not, just "Was reading the code and just wanted to make it more defensive".

**Comment by ashcrow** [comment]
@cgwalters being more defensive. I want to add some catching around other possible errors as well to allow for some stronger logging. I figured I'd start with `json`.

I force pushed the commit without the full stop.

File: homu/main.py (hunk 1 above)
Comment (ashcrow) [inline homu/main.py]: Updated in fixup.

**Pull request closed without merging**